rinja/src/filters/humansize.rs

```diff
@@ -21,24 +61,97 @@ use humansize::{DECIMAL, ISizeFormatter, ToF64};
 /// assert_eq!(tmpl.to_string(),  "Filesize: 1.23 MB.");
 /// ```
 #[inline]
-pub fn filesizeformat(b: &impl ToF64) -> Result<FilesizeFormatFilter, Infallible> {
-    Ok(FilesizeFormatFilter(b.to_f64()))
+pub fn filesizeformat<T: ToF32>(b: T) -> Result<FilesizeFormatFilter, Infallible> {
+    Ok(FilesizeFormatFilter(b.to_f32()))
 }
 
 #[derive(Debug, Clone, Copy)]
-pub struct FilesizeFormatFilter(f64);
+pub struct FilesizeFormatFilter(f32);
+
+struct NbAndDecimal(usize);
+impl NbAndDecimal {
+    fn new(nb: f32) -> Self {
+        // `nb` will never be bigger than 999_999 so we're fine with usize.
+        Self(nb as _)
+    }
+}
```

Comment:
I would explicitly use `NbAndDecimal(u32)`: `usize` would be too small on 16bit systems, and on x64 sytems, working with `u64` is a lot slower than `u32`.

Reply:
I wondered about that but I guess you're right.